client/my-sites/checkout/composite-checkout/test/checkout-contact-step.js

```diff
@@ -269,12 +265,13 @@ describe( 'Checkout contact step', () => {
 		render( <MyCheckout cartChanges={ cartChanges } />, container );
 		// Wait for the cart to load
 		await screen.findByText( 'Country' );
-		// Wait for the validation to complete
-		await waitForElementToBeRemoved( () => screen.queryAllByText( 'Please wait…' ) );
-		expect( screen.queryByTestId( 'payment-method-step--visible' ) ).toBeInTheDocument();
+		// Wait for the validation to complete, then check we've advanced to the next payment step
+		await waitFor( () =>
+			expect( screen.queryByTestId( 'payment-method-step--visible' ) ).toBeInTheDocument()
+		);
 	} );
 
-	it.skip( 'does not autocomplete the contact step when there are invalid cached details', async () => {
+	it( 'does not autocomplete the contact step when there are invalid cached details', async () => {
```

Comment:
If we fix up the format of the mocked endpoint response message, we should be able to query for the message in the page.

```
	it( 'does not autocomplete the contact step when there are invalid cached details', async () => {
		mockCachedContactDetailsEndpoint( {
			country_code: 'US',
			postal_code: 'ABCD',
		} );
		mockContactDetailsValidationEndpoint( 'tax', {
			success: false,
			messages: {
				postal_code: [ 'Make sure this field is valid.' ],
			},
		} );
		const cartChanges = { products: [ planWithoutDomain ] };
		render( <MyCheckout cartChanges={ cartChanges } />, container );
		// Wait for the cart to load
		await screen.findByText( 'Country' );
		// Wait for the validation message to show
		await screen.findByText( 'Make sure this field is valid.' );
		// Check that we never advance to the next payment step
		await expect( screen.findByTestId( 'payment-method-step--visible' ) ).toNeverAppear();
	} );
```